CMakeLists.txt

```diff
@@ -1,303 +1,159 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.12)
-PROJECT(ARRAYFIRE)
+cmake_minimum_required(VERSION 3.5.2)
+project(ArrayFire
+  VERSION 3.5.0
+  LANGUAGES C CXX )
 
-SET_PROPERTY(GLOBAL PROPERTY USE_FOLDERS ON)
-if(POLICY CMP0058)
-  CMAKE_POLICY(SET CMP0058 NEW)
-endif(POLICY CMP0058)
-
-SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")
-INCLUDE(UploadCoveralls)
-INCLUDE(AFInstallDirs)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")
 
-OPTION(BUILD_TEST "Build Tests" ON)
-OPTION(BUILD_EXAMPLES "Build Examples" ON)
+include(CMakeDependentOption)
 
-OPTION(BUILD_CPU "Build ArrayFire with a CPU backend" ON)
+find_package(CUDA 7.0)
```

Comment:
@umar456 I sometimes disable `BUILD_CUDA` when I want to build CPU quickly. We should still have the ability to disable any backend on demand.